packages/theme/src/cli/utilities/theme-environment/hot-reload/server.ts

```diff
@@ -81,7 +78,7 @@ export function getInMemoryTemplates(ctx: DevServerContext, currentRoute?: strin
       } else if (!fileKey.includes('.default.')) continue
     }
 
-    inMemoryTemplates[fileKey] = content
+    inMemoryTemplates[fileKey] = ctx.localThemeFileSystem.files.get(fileKey)?.value ?? ''
```

Comment:
Q for curiosity: What would happen when we delete the file locally and are waiting for the remote file to be deleted?

I believe `files.get(fileKey)` would be undefined in this case. Do we still want to return an empty string? Would an early return work as well?

Reply:
>I believe files.get(fileKey) would be undefined in this case

Exactly, since it's not removed in the remote renderer yet, we pass empty string from here to make it behave at least like an empty file. For example, this way the remote renderer won't render sections that have been removed locally, even if the section file still exists in the remote.